rust/src/coordinator.rs

```diff
@@ -116,10 +121,15 @@ impl Coordinator {
 
     /// Validate and handle an update message.
     fn handle_update_message(&mut self, bytes: &[u8]) -> Result<(), PetError> {
-        let msg = UpdateMessage::open(bytes, &self.pk, &self.sk, self.sum_dict.len())?;
+        let msg = UpdateMessage::open(bytes, &self.pk, &self.sk)?;
+        if msg.local_seed_dict().keys().collect::<HashSet<_>>()
```

Comment:
For the future, we should try to see if there's a cheaper way to do that. This can quickly become expensive if there are many participants.

Reply:
an equivalent way to compare the key sets would be
```
if dict1.len() == dict2.len() && dict1.keys().all(|key| dict2.contains_key(key)) {
    // do sth
} else {
    // do other
}
```
this might me more efficient due to boolean operations only.

Reply:
and i've just seen a built-in way to that even more efficiently:
```
if dict1.keys.eq(dict2.keys()) {
    // do sth
} else {
    // do other
}
```
that's probably the way to  go.

Reply:
Are we guaranteed that the two iterators produced by `.keys()` iterate through the keys in the same order? Otherwise it's not strictly equivalent.

Reply:
i just tested this and unfortunately the `eq()` depends on the order whereas `keys()` produces an iterator of arbitrary order. i guess we have to go with the other approach then.